### PR TITLE
Fix published Svelte hydration identity

### LINKED ISSUE #746: Chat emits hydration mismatch for an empty SSR conversation
## Summary

`@lostgradient/cinder@0.13.0` emits a Svelte `hydration_mismatch` when `Chat` hydrates an empty conversation in a SvelteKit SSR route.

## Reproduction

Render Cinder Chat inside a server-rendered SvelteKit route with a stable id and an empty conversation:

```svelte
<script lang="ts">
  import Chat from "@lostgradient/cinder/chat";
  const conversation = { id: "session-example", messages: [] };
</script>

<Chat id="session-example" {conversation} />
```

Run the route through Vite development SSR and attach a browser console listener before navigation. Hydration logs:

```text
[svelte] hydration_mismatch
Hydration failed because the initial UI does not match what was rendered on the server
```

Downstream reproduction: `stevekinney/stardust` issue #39 on Cinder 0.13.0, Svelte 5.56.x, SvelteKit 2.68.x. The downstream wrapper uses a stable `session-e2e-existing-session` id and an empty event transcript.

## Isolation evidence

The session route fails its hydration smoke test with Cinder `Chat` rendered inside the Stardust `ConversationView`. Removing only the `Chat` node makes the same route pass while leaving the Stardust wrapper mounted. Gating the whole conversation until mount also suppresses the warning, but that is a downstream workaround and should not be required.

Current Chat SSR coverage checks module evaluation without DOM globals and source-level virtualization gating. It does not perform a full server-render-and-hydrate round trip of the Chat tree.

## Expected behavior

A stable empty conversation must hydrate without warnings, preserving the server-rendered empty state and composer.

## Acceptance criteria

- Add a full SSR-render-and-hydrate regression test for Chat with a stable id and empty conversation.
- Fix the server/client DOM divergence without requiring consumers to disable SSR or defer Chat until mount.
- Preserve empty-state, composer, live-region, and virtualization behavior.

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

- normalize package component compiler filenames so staged server bundles and published client source generate identical scoped CSS hashes
- extend the hydration helper to server-build complete local Svelte component graphs and add an empty Chat regression test
- exercise the packed empty Chat route in both development and production SvelteKit consumer validation

## Verification

- `bun run validate`
- pre-commit and pre-push hooks

Closes #746

### CODE REVIEW

File: packages/components/src/test/hydrate.ts [hunk 1]
```diff
@@ -216,12 +309,14 @@ export async function renderThenHydrate<Props extends Record<string, unknown>>(
...
-      unlinkSync(file);
-    } catch {
-      // Already gone — ignore.
+    for (const tempFile of [file, serverRuntimeShimPath]) {
+      try {
+        unlinkSync(tempFile);
+      } catch {
+        // Already gone — ignore.
+      }
+      pendingTempFiles.delete(tempFile);
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/test/hydrate.ts]: Fixed in ce200752. Temp-file deregistration now occurs only after successful unlink or ENOENT; other removal failures remain registered for the exit safety net. Added a regression test covering both EACCES retention and ENOENT deregistration.

File: packages/components/src/test/hydrate.ts [hunk 2]
```diff
@@ -170,7 +247,36 @@ export async function renderThenHydrate<Props extends Record<string, unknown>>(
...
+        'export async function tick() {}',
+        'export async function settled() {}',
+        "export function mount() { e.lifecycle_function_unavailable('mount'); }",
+        "export function hydrate() { e.lifecycle_function_unavailable('hydrate'); }",
+        "export function unmount() { e.lifecycle_function_unavailable('unmount'); }",
+        "export function fork() { e.lifecycle_function_unavailable('fork'); }",
+        '',
+      ].join('\n'),
+    );
+    let serverCode = await artifact.text();
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/test/hydrate.ts:272]: Fixed in 15558138. The generated SSR module is now imported through `pathToFileURL(file).href`, matching `server-render.ts` and producing a spec-compliant module URL on Windows. The focused hydration suite passes (5 tests), and the package lint/typecheck/pre-push gates are green.
